src/java.compiler/share/classes/javax/lang/model/util/package-info.java

```diff
@@ -35,6 +35,274 @@
  * <p> Unless otherwise specified, methods in this package will throw
  * a {@code NullPointerException} if given a {@code null} argument.
  *
+ * @apiNote
+ *
+ * <h2 id=expectedEvolution>Expected visitor evolution</h2>
+ *
+ * As the Java programming language evolves, the visitor interfaces of
+ * the language model also evolve as do the concrete visitors in this
+ * package. A preview language feature in JDK <i>N</i> may have API
+ * elements added in the set of visitors for the preview language
+ * level. Such new elements are marked as reflective preview API. Any
+ * existing methods whose specification is updated to support the
+ * preview feature are <em>not</em> marked as preview.
+ *
+ * <p>The remainder of this note will show two examples of the API
+ * changes in the model and visitors that can be added to support a
+ * language feature. The examples will use additions to the elements
+ * portion of the language model, but the updates to visitors for
+ * types or annotation values would be analogous.
+ *
+ * Two distinct cases are:
+ * <ul>
+ *
+ * <li>the preview language construct has a corresponding new modeling
+ * interface
+ *
+ * <li>the preview language construct only triggers the introduction
+ * of a new kind <em>without</em> a new modeling interface.
```

Comment:
There are many kinds of kind. Can you qualify which kind you mean here?

Reply:
Add some more explicit information; thanks for the suggestion.